### ISSUE TITLE
Have the bot respond to bug urls

### ISSUE THREAD

**Issue by bobsilverberg** [davehunt/hubot-bugzilla#2]
## Sample Interaction

```
user1>> I love pasting bug urls, like this one https://bugzilla.mozilla.org/show_bug.cgi?id=689423
hubot>> Bug 689423 normal, NEW, Add Angry Birds to Firefox new tab page
```

**Comment by bobsilverberg** [comment]
I spoke to glob about how the current bots do this, and was informed that they listen for emails, parse the email, write to a file, and then the bot reads that file to issue messages. The source code for the current bot (written in perl) can be found at [1]. He further suggested that if we are only interested in new bugs and status changes (and not flag changes), we probably don't need to parse the email text at all - we can probably just use the email headers. We could also consider just grabbing the bug number from the `x-bugzilla-id` header and then look up the bug on bmo.

[1] https://github.com/globau/firebot/blob/master/BotModules/BugzillaMailHandler.pl

**Comment by davehunt** [comment]
It sounds like you're talking about announcing new bugs for a component in the channel, but this issue is about responding to a bug URL mentioned in the channel. Did you mean to put this information here?

**Comment by bobsilverberg** [comment]
You're right! This is not the right place for this. I guess we should open a separate issue for this feature, which only existed in my head until now.

**Comment by bobsilverberg** [comment]
@davehunt I already miss this! Do you think you might be able to find time to address this anything soon? I tried for a few minutes, but my regex chops are totally not up to the task!

**Comment by davehunt** [comment]
The difficulty will be making sure the URL is recognised from the environment variable. I did already spend some time on this before abandoning it for the initial release. I don't think I'll have time to work on it anytime soon. Did you want to consider using one of the other bug bots for this?

**Comment by bobsilverberg** [comment]
I might be misunderstanding, but it seems to me that that part is easy. Wouldn't it just be something like:

`theRegex = new RegExp("startOfRegex" + urlFromEnvironmentVar + "endOfRegex")`

What I'm not sure about is the values for `startOfRegex` and `endOfRegex`.  Perhaps I'll try again tomorrow.

**Comment by davehunt** [comment]
Well the `urlFromEnvironmentVar` would need to be escaped for a start. Patches are welcome, this is not really a feature I use, so I don't miss it personally. :smile:
